portfolio/src/main/java/com/google/sps/servlets/CommentsServlet.java

```diff
@@ -49,10 +55,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery commentResults = datastore.prepare(commentsQuery);
 
+    /**
+     * Apply limit if necessary. 
+     * Use offset 0 so that we can have a default options if try/catch fails.
+     */
+    FetchOptions options = FetchOptions.Builder.withOffset(0);
+    try {
+        Integer limit = Integer.parseInt(commentsCount);
+        options = options.limit(limit);
+    } catch (NumberFormatException expectedIfNoLimit){
+        /** Don't set limit if not needed. */
```

Comment:
While this may fly in the face of the starter project guidance, it's a good practice to always have a limit, even if that limit is large.  Whenever you have code that tries to load everything from a database you have the risk that it'll break either the server or the user experience (or both).

Reply:
That makes sense, would you say its alright to hard-cap it, or only have a cap if you can choose the start-element too?